### PR TITLE
IW4Of Fixes

### PR DESCRIPTION
**Pull request by Rackover**
Some stuff we forgot or messed up

### CODE REVIEW

**Comment by Rackover** [comment]
Still not building valid zones btw so wait a bit before merging, it's not over

File: src/Components/Modules/ZoneBuilder.cpp [hunk 1]
```diff
@@ -795,9 +795,16 @@ namespace Components
 			}
 		};
 
-		params.work_directory = (*Game::fs_basepath)->current.string;
+		if (Game::fs_basepath && Game::fs_gameDirVar)
```
Comment (diamante0018) [inline src/Components/Modules/ZoneBuilder.cpp:798]: this is wrong it should be *fs_basepath && *fs_gameDirVar
the type of this variable is ** (double pointer) so this if statement is never false

**Review by diamante0018 — CHANGES_REQUESTED** [review]

File: src/Components/Modules/ZoneBuilder.cpp [hunk 2]
```diff
@@ -535,10 +535,11 @@ namespace Components
...
 	void ZoneBuilder::Zone::addBranding()
 	{
-		constexpr auto* data = "Built using the IW4x Zone:B:uilder Version 4";
-		auto dataLen = std::strlen(data); // + 1 is added by the save code
+		const auto now = std::chrono::system_clock::now();
+		std::string data = std::format("Built using the IW4x ZoneBuilder! {:%d-%m-%Y %H:%M:%OS}", now);
+		auto dataLen = data.size(); // + 1 is added by the save code
 
-		this->branding = {this->zoneName.data(), 0, static_cast<int>(dataLen), data};
+		this->branding = {this->zoneName.data(), 0, static_cast<int>(dataLen), data.data()};
```
Comment (diamante0018) [inline src/Components/Modules/ZoneBuilder.cpp:542]: pretty sure UAF for data, have you tested this?

Comment (Rackover) [reply src/Components/Modules/ZoneBuilder.cpp:542]: Not yet this is a WIP, i have bigger problem right now with remappedtechinquesets not being saved properly, I'll come back to this before this PR is ready :)

Comment (diamante0018) [reply src/Components/Modules/ZoneBuilder.cpp:542]: Okay, I'll mark this as "request changes" if I can so it does not go unforgotten. this becomes a nulltpr after the function exits.

File: src/Components/Modules/Maps.cpp [hunk 3]
```diff
@@ -666,12 +666,24 @@ namespace Components
 		return Utils::Hook::Call<bool(Game::gentity_s*)>(0x5050C0)(ent);
 	}
 
-	int16 Maps::CM_TriggerModelBounds(int modelPointer, Game::Bounds* bounds) {
-#ifdef DEBUG
+	int16 Maps::CM_TriggerModelBounds(unsigned int triggerIndex, Game::Bounds* bounds) {
```
Comment (diamante0018) [inline src/Components/Modules/Maps.cpp:669]: Please fix the braces and rename function to contain the word Stub or Hk or Hook

File: src/Components/Modules/ZoneBuilder.cpp [hunk 4]
```diff
@@ -535,10 +535,11 @@ namespace Components
 	// Add branding asset
 	void ZoneBuilder::Zone::addBranding()
 	{
-		constexpr auto* data = "Built using the IW4x Zone:B:uilder Version 4";
-		auto dataLen = std::strlen(data); // + 1 is added by the save code
+		const auto now = std::chrono::system_clock::now();
+		std::string data = std::format("Built using the IW4x ZoneBuilder! {:%d-%m-%Y %H:%M:%OS}", now);
```
Comment (diamante0018) [inline src/Components/Modules/ZoneBuilder.cpp:539]: TBH I didn't know this was possible but I tested it and it works.
Example output: `Built using the IW4x ZoneBuilder! 16-02-2023 18:51:33`
Just use auto please.

Comment (Rackover) [reply src/Components/Modules/ZoneBuilder.cpp:539]: Ok so no need to fix UAF or anything ?

Comment (diamante0018) [reply src/Components/Modules/ZoneBuilder.cpp:539]: What? This is unrelated

Comment (diamante0018) [reply src/Components/Modules/ZoneBuilder.cpp:539]: UAF is still present a couple of lines after this

Comment (Rackover) [reply src/Components/Modules/ZoneBuilder.cpp:539]: Did not know what you were talking about with "i tested it and it works"

Comment (diamante0018) [reply src/Components/Modules/ZoneBuilder.cpp:539]: My bad. What I meant to say is that I didn't know you could format the time like that. But yeah the UAF is present.
🅱️

**Comment by Rackover** [comment]
This is ready as far as i'm concerned, builds maps properly

**Review by diamante0018 — DISMISSED** [review]

**Review by diamante0018 — APPROVED** [review]

**Pull request merged**